tests/test_polling_manager.py

```diff
@@ -40,31 +42,47 @@ def test_polling_manager_calls_update_environment_on_each_refresh():
     polling_manager.stop()
 
 
-def test_polling_manager_is_resilient_to_api_errors(mocker, server_api_key):
+@responses.activate()
+def test_polling_manager_is_resilient_to_api_errors(
+    flagsmith: Flagsmith,
+    environment_json: str,
+    mocker: MockerFixture,
+    server_api_key: str,
+) -> None:
     # Given
-    session_mock = mocker.patch("requests.Session")
-    session_mock.get.return_value = mock.MagicMock(status_code=500)
+    responses.add(method="GET", url=flagsmith.environment_url, body=environment_json)
     flagsmith = Flagsmith(
         environment_key=server_api_key,
         enable_local_evaluation=True,
         environment_refresh_interval_seconds=0.1,
     )
+
+    session_mock = mocker.patch("requests.Session")
+    session_mock.get.return_value = mock.MagicMock(status_code=500)
```

Comment:
Since we're using `responses` here, why not just add a second response that raises an exception? e.g. [here](https://github.com/getsentry/responses?tab=readme-ov-file#exception-as-response-body) or: 

```python
responses.add(method="GET", url=flagsmith.environment_url, status=500)
```

Reply:
Sure that's easy. Done.